src/main/java/ru/mercuriev/game/of/life/graph/cell/CellBuilder.java

```diff
@@ -0,0 +1,179 @@
+package ru.mercuriev.game.of.life.graph.cell;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * This stands for object which:
+ * 1. holds the pointer to the current cell ({@link Cell}) while constructing new Graph
+ * 2. appends new cells to the graph
+ * 3. merging together current and specified CellBuilder
+ * Each time any method of the object is called current cell is the right and bottom most cell in graph (*1*)
+ *
+ * Class is not thread safe - do not use it without proper synchronization (e.g. in parallel stream )
+ *
+ * @author paul
+ */
+final class CellBuilder {
+
+    /**
+     * current cell
+     */
+    private Cell cell = null;
```

Comment:
This POV is controversial. My understaing of what is current cell for a method can be illustrated by `public Optional<Cell> build()` method

And CellBuilder's `private Cell cell = null;` stands for 'this builder's current cell'

I agree, conbination of the field name and its comment is bad. But renaming it to the currentCell will amke the situation even worse, i think. Maybe you'll suggest more concise comment?